lib/rubocop/cop/rspec/let_setup.rb

```diff
@@ -45,10 +45,19 @@ def on_block(node)
         private
 
         def unused_let_bang(node)
-          let_bang(node) do |method_send, method_name|
+          child_let_bang(node) do |method_send, method_name|
             yield(method_send) unless method_called?(node, method_name)
           end
         end
+
+        def child_let_bang(block_node)
+          return unless block_node.body&.begin_type?
+
+          block_node.body.each_child_node(:block) do |child|
```

Comment:
It would be great if you could `yield` if the passed block yielded, e.g. if there is shorter or more FP way to write the code below

Reply:
Do you mean something like this?

```ruby
block_node.body.each_child_node(:block)
  .map { |child| let_bang(child) }
  .reject { |m| m.nil? }
  .each { |m| yield m }
```

Reply:
yes, or
```
block_node.body.each_child_node(:block)
  .map(&method(:let_bang)
  .compact
  .each { |m| yield m }
```
But would prefer to have it even shorter, or at least without having to use explicit blocks
Ideally something like
`block_node.body.each_child_node(:block).yield_when(&method(:let_bang)`

Anyway, that was just a side note, as I hate to see things like
```
a = something
yield a if a
```

Reply:
Sorry, I've missed that this has been discussed already.